pages/balance-budget.vue

```diff
@@ -267,31 +183,79 @@ import FiscalYearSelect from "@/components/FiscalYearSelect";
+  },
+  mounted() {
+      this.is_mounted = true;
+      this.exceeds_limit = false;
   },
   data() {
-    return {
-      selected_city: "San Francisco",
-      selected_year: "2020-2021",
-      cities: ["San Francisco", "Oakland"],
-      years: [
-        "2015-2016",
-        "2016-2017",
-        "2017-2018",
-        "2018-2019",
-        "2019-2020",
-        "2020-2021"
-      ]
-    };
+        return {
+            total_expenses: total_expenses,
+            selected_city: "San Francisco",
+            selected_year: "2020-2021",
+            cities: ["San Francisco", "Oakland"],
+            years: ["2015-2016","2016-2017","2017-2018","2018-2019","2019-2020","2020-2021"],
+            is_mounted: false,
+            budget_pie_chart_data: [
+                {name: "Community Health", total: starting_expense, dept_color: '#2A6465'},
+                {name: "Culture & Recreation", total: starting_expense, dept_color: '#EF896E'},
+                {name: "General Admin & Finance", total: starting_expense, dept_color: '#F5BD41'},
+                {name: "General City Responsibilities", total: starting_expense, dept_color: '#CAAA97'},
+                {name: "Human Welfare & Neighborhood Development", total: starting_expense, dept_color: '#4DA54A'},
+                {name: "Public Protection", total: starting_expense, dept_color: '#4296AD'},
+                {name: "Public Works, Transportation & Commerce", total: starting_expense, dept_color: '#CF722A'}
+            ],
+            budget_pie_chart_config: {
+                key: "name",
+                value: "total",
+                color: {key: "dept_color"},
+                margin: {left: 100, right: 100},
+                transition: {duration: 100, ease: "easeLinear"}
+            },
+            slider_max: total_expenses,
+            slider_min: 0,
+            health_value: starting_expense,
+            culture_value: starting_expense,
+            admin_value: starting_expense,
+            city_value: starting_expense,
+            welfare_value: starting_expense,
+            protection_value: starting_expense,
+            transport_value: starting_expense,
+            test_rule: [v => current_sum_expenses <= total_expenses]
+        }
+  },
+  methods: {
+      editPieChart(dept, dept_value, color) {
+          var old_value = this.budget_pie_chart_data[this.budget_pie_chart_data.findIndex(({name}) => name == dept)].total
+          current_sum_expenses -= old_value
+          current_sum_expenses += dept_value
+          console.log(current_sum_expenses)
+          console.log(total_expenses)
+          if(current_sum_expenses > total_expenses){
+              this.exceeds_limit = true
+          } else {
+              this.exceeds_limit = false
+          }
+
+          this.budget_pie_chart_data.splice(this.budget_pie_chart_data.findIndex(({name}) => name == dept),1)
+          this.budget_pie_chart_data.push({name: dept, total: dept_value, dept_color: color})
```

Comment:
one issue with this approach is that it changes the order of departments within the pie chart if you jump around when editing sliders, which i imagine could be confusing. It's also a little strange to be passing around colors -- I think finding the existing pie slice and editing the total amount would be a better pattern.

I think this will implicitly be fixed by refactoring state into the vuex model that I was talking about though, so don't worry about it!

Reply:
agreed that passing through the colors + having the ordering change here is a bit odd! this was my workaround as I was struggling to have the d3 chart refresh from a change in underlying data for an existing pie slice 

will leave as is for now, but will re-visit later!